src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -377,30 +378,18 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
         LOG_VERB("Trying to preemptively set tensors for Subgraph[" << i << "]...");
         LOG_BLOCK();
         auto& comp_model_desc = m_npuw_model->m_compiled_submodels[i];
-
         if (!comp_model_desc.compiled_model || !comp_model_desc.replaced_by) {
             continue;
         }
-
         const auto real_idx = comp_model_desc.replaced_by.value();
         auto& func_desc = m_npuw_model->m_compiled_submodels[real_idx];
 
-        auto& request = m_subrequests[real_idx];
-
-        for (std::size_t cidx = 0u; cidx < comp_model_desc.closure.size(); cidx++) {
-            const auto& closure = comp_model_desc.closure[cidx];
-
-            const auto closure_param_id = comp_model_desc.param_base + cidx;
-            const auto& iport = func_desc.compiled_model->inputs()[closure_param_id];
-
-            // No update required to this tensor in runtime - so it can be set only once
-            // Will be utilized when there is no FOLDing
-            if (!m_npuw_model->m_update_required) {
-                // At this point closure already contains allocated and transformed tensor ready to be used
-                request->set_tensor(iport, ov::get_tensor_impl(closure));
-            }
-        }  // for(closure)
-        LOG_VERB("DONE");
+        // So - closure update is NOT required, OR the function is SINGLE -
+        // just handle it's closure here and don't do it in runtime
+        if (!m_closure_update_required || func_desc.forced_to_fcall) {
+            unpack_closure(i, m_subrequests[real_idx]);
```

Comment:
So it'll be called in CWAI case or in forced case. Why do we change `set_tensor` to `unpack_closure` for CWAI case?

Reply:
unpack_closure either unpacks or sets the closure. In CWAI case, no unpacks should be done. Assume calling `unpack_closure` is a generic way to initialize the funcall's closure